www/netlify/functions/trackCLICommands.js

```diff
@@ -0,0 +1,23 @@
+const { v4: uuidv4 } = require('uuid');
+const Analytics = require('analytics-node');
+
+const analytics = new Analytics(process.env.SEGMENT_KEY);
+
+exports.handler = async function eventHandler(event) {
+  // Only allow POST
+  if (event.httpMethod !== 'POST') {
+    return { statusCode: 405, body: 'Method Not Allowed' };
+  }
+  const { eventName } = JSON.parse(event.body);
+  // dispatch event to Segment
+  analytics.track({
+    anonymousId: uuidv4(),
+    event: 'openedx.paragon.functions.track-cli-commands.run',
+    properties: { eventName },
+  });
+
+  return {
+    statusCode: 200,
+    body: JSON.stringify({ success: true }),
+  };
+};
```

Comment:
I don't think you need to create another netlify function here, instead we should modify the one we have, because they do the same thing.

I would do the following
1. Rename this file to `sendAnalyticsData.js` and modify the function as follows:
```js
exports.handler = async function eventHandler(event) {
  // Only allow POST
  if (event.httpMethod !== 'POST') {
    return { statusCode: 405, body: 'Method Not Allowed' };
  }
  const { eventId, properties } = JSON.parse(event.body);
  // dispatch event to Segment
  analytics.track({
    anonymousId: uuidv4(),
    event: eventId,
    properties,
  });

  return {
    statusCode: 200,
    body: JSON.stringify({ success: true }),
  };
};
```
2. Keep `trackGenerateComponent.js` in order not to make a breaking change, old version of Paragon will still send data to this function, so wee need to keep this functionality for now. Although, we probably could reuse our new handler here to avoid code duplication, something like this (not sure if it will work):
```js
const { handler: actualHandler } = require('./sendAnalyticsData');

exports.handler = async function eventHandler(event) {
  const body = JSON.parse(event.body);
  event.body = JSON.stringify({
    ...body,
    eventId: 'openedx.paragon.functions.track-generate-component.created',
    properties: { componentName: body.componentName }
  })
  
  return actualHandler(event);
};
```

3. Modify `sendTrackInfo` function to
```js
function sendTrackInfo(eventId, properties) {
  const { BASE_URL, TRACK_ANONYMOUS_ANALYTICS } = process.env;
  if (TRACK_ANONYMOUS_ANALYTICS) {
    const url = `${BASE_URL}/.netlify/functions/sendTrackData`;
    axios.post(url, { eventId, properties })
      .then(result => {
        // eslint-disable-next-line no-console
        console.log(`Track info is successfully sent (status ${result.status})`);
      }).catch(error => {
        // eslint-disable-next-line no-console
        console.log(`Track info request failed (${error})`);
      });
  }
}
```

Then you could easily use this one function for any event you want, e.g. to track component generation you would do:
```js
sendTrackInfo('openedx.paragon.functions.track-generate-component.created', { componentName });
```

to track CLI commands:
```js
sendTrackInfo('openedx.paragon.cli-command.used', { command });
```

basically it will be able to handle any event you want in the future also without any modifications.

Reply:
Done